src/vs/code/electron-sandbox/workbench/workbench.ts

```diff
@@ -90,15 +90,19 @@
 			splash.className = baseTheme ?? 'vs-dark';
 
 			if (layoutInfo.windowBorder && colorInfo.windowBorder) {
-				splash.style.position = 'relative';
-				splash.style.height = 'calc(100vh - 2px)';
-				splash.style.width = 'calc(100vw - 2px)';
-				splash.style.border = `1px solid var(--window-border-color)`;
-				splash.style.setProperty('--window-border-color', colorInfo.windowBorder);
+				const borderElement = document.createElement('div');
```

Comment:
This fixes an issue where on macOS the border was broken at the rounded corners. Making the border its own element above others addresses it.